quasar-core/src/main/java/co/paralleluniverse/fibers/instrument/CheckInstrumentationVisitor.java

```diff
@@ -154,10 +154,9 @@ public MethodVisitor visitMethod(final int access, final String name, final Stri
 
                 @Override
                 public AnnotationVisitor visitAnnotation(String adesc, boolean visible) {
-                    super.visitAnnotation(adesc, visible);
                     if (adesc.equals(SUSPENDABLE_DESC))
                         susp = true;
-                    return null;
+                    return super.visitAnnotation(adesc, visible);
```

Comment:
Returning `null` stops visit proceeding whereas it will now visit the reset of the annotation's byte-code. Why are you changing this in the first place? Is there an issue here?

Reply:
There is no issue at the moment. But the visitAnnotation calls in quasar are inconsistent at best. Some return null and some call the super class. So I was trying to be consistent with other classes where scanning is done and the super is called. The default implementation of which returns null anyway. You'll also notice that in visitEnd in that class the super is called. So why not be consistent and do the same here? It just seemed odd to me.

Reply:
It's all down to how the ASM "Visitor" API chains visitors together. `visitEnd()` returns `void` and every visitor in the chain needs to invoke it, whereas `visitAnnotation()`'s return value controls whether or not ASM tries to visit all of the elements of the annotation. If you know what the annotation is (i.e. has no elements / nested annotations) then returning `null` could be an optimisation. However, you need to be careful because returning `null` is also how you tell ASM to delete things....